src/lib.rs

```diff
@@ -121,41 +121,71 @@ use std::env;
 use std::path::{Path, PathBuf};
 use std::process::Command;
 use std::str::from_utf8;
+use std::fmt;
+
+use semver::Version;
 
 pub use errors::{Error, ErrorKind, Result};
 pub use dependency::{Dependency, DependencyKind};
 
 mod errors;
 mod dependency;
 
+/// An "opaque" identifier for a package.
+/// It is possible to inspect the `repr` field, if the need arises, but its
+/// precise format is an implementation detail and is subject to change.
+///
+/// `Metadata` can be indexed by `PackageId`.
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[serde(transparent)]
+pub struct PackageId {
+    /// The underlying string representation of id.
+    pub repr: String
+}
```

Comment:
This move lost the convenience methods that existed on `WorkspaceMember`

Reply:
Oh nevermind. I should have read the issue first

Reply:
Yep, they are now basically `metadata[package_id].url` etc